test/Common/TestOrchestrations.cs

```diff
@@ -28,6 +28,55 @@ public static async Task<string> SayHelloWithActivity([OrchestrationTrigger] Dur
             return output;
         }
 
+        public static async Task<bool> SayHelloWithActivityWithDeterministicGuid([OrchestrationTrigger] DurableOrchestrationContext ctx)
+        {
+            string input = ctx.GetInput<string>();
+            Guid firstGuid = ctx.NewGuid();
+            Guid secondGuid = ctx.NewGuid();
+            string output = await ctx.CallActivityAsync<string>(nameof(TestActivities.Hello), input);
+            Guid thirdGuid = ctx.NewGuid();
+            return firstGuid != secondGuid && firstGuid != thirdGuid && secondGuid != thirdGuid;
+        }
+
+        public static bool VerifyUniqueGuids([OrchestrationTrigger] DurableOrchestrationContext ctx)
+        {
+            List<Guid> guids = new List<Guid>(10000);

```

Comment:
A `HashSet<Guid>` would be much more natural and efficient than `List<Guid>` for a test like this.

Reply:
Great!

I will change it!

Thank you!